api/app/controllers/webhook.controller.ts

```diff
@@ -319,16 +342,29 @@ export async function handlePushEvent(req: Request, res: Response) {
       .json({ message: "Invalid branch for the current environment" });
   }
 
+  let processedCommits = 0;
+
   try {
-    for (const commit of commits) {
-      await processCommit(commit, pushEvent, "added", branch);
-      await processCommit(commit, pushEvent, "modified", branch);
-    }
+    await Promise.all(
+      commits.map(async (commit: any) => {
+        try {
+          await processCommit(commit, pushEvent, "added", branch);
+          await processCommit(commit, pushEvent, "modified", branch);
+          processedCommits++;
```

Comment:
If we have commit A,B,C and B throws an error inside of `processCommit` then `processedCommits` will be equal to 1, but B was also processed - its processing resulted in an error. So the correct number is 2 processed commits, therefore 1 unprocessed.